docs/core/whats-new/dotnet-10/runtime.md

```diff
@@ -2,15 +2,90 @@
 title: What's new in .NET 10 runtime
 description: Learn about the new features introduced in the .NET 10 runtime.
 titleSuffix: ""
-ms.date: 06/09/2025
+ms.date: 07/16/2025
 ms.topic: whats-new
 ai-usage: ai-assisted
 ---
 # What's new in the .NET 10 runtime
 
-This article describes new features and performance improvements in the .NET runtime for .NET 10. It's updated for Preview 5.
+This article describes new features and performance improvements in the .NET runtime for .NET 10. It has been updated for Preview 6.
 
-## Array interface method devirtualization
+## JIT compiler improvements
+
+The JIT compiler in .NET 10 includes significant enhancements that improve performance through better code generation and optimization strategies.
+
+### Improved code generation for struct arguments
+
+.NET's JIT compiler is capable of an optimization called physical promotion, where the members of a struct are placed in registers rather than on the stack, eliminating memory accesses. This optimization is particularly useful when passing a struct to a method, and the calling convention requires the struct members to be passed in registers.
+
+.NET 10 improves the JIT compiler's internal representation to handle values that share a register. Previously, when struct members needed to be packed into a single register, the JIT would store values to memory first and then load them into a register. Now, the JIT compiler can place the promoted members of struct arguments into shared registers directly, eliminating unnecessary memory operations.
+
+Consider the following example:
+
+```csharp
+struct Point
+{
+    public long X;
+    public long Y;
+
+    public Point(long x, long y)
+    {
+        X = x;
+        Y = y;
+    }
+}
+
+[MethodImpl(MethodImplOptions.NoInlining)]
+private static void Consume(Point p)
+{
+    Console.WriteLine(p.X + p.Y);
+}
+
+private static void Main()
+{
+    Point p = new Point(10, 20);
+    Consume(p);
+}
+```
+
+On x64, we pass the members of `Point` to `Consume` in separate registers, and since physical promotion kicked in for the local `p`, we don't allocate anything on the stack first:
+
+```asm
+Program:Main() (FullOpts):
+       mov      edi, 10
+       mov      esi, 20
+       tail.jmp [Program:Consume(Program+Point)]
+```
+
+Now, suppose we changed the members of `Point` to be `ints` instead of `longs`. Because `ints` are four bytes wide, and registers are eight bytes wide on x64, the calling convention requires us to pass the members of `Point` in one register. Previously, the JIT compiler would first store the values to memory, and then load the eight-byte chunk into a register. With the .NET 10 improvements, the JIT compiler can now place the promoted members of struct arguments into shared registers directly:
```

Comment:
Let's fix the code fences. The "s" isn't a keyword:

```suggestion
Now, suppose we changed the type of the members of `Point` to `int` instead of `long`. Because an `int` is four bytes wide, and registers are eight bytes wide on x64, the calling convention requires us to pass the members of `Point` in one register. Previously, the JIT compiler would first store the values to memory, and then load the eight-byte chunk into a register. With the .NET 10 improvements, the JIT compiler can now place the promoted members of struct arguments into shared registers directly:
```